cloudify_agent/installer/operations.py

```diff
@@ -61,13 +64,22 @@ def create(cloudify_agent, installer, **_):
         cloudify_agent['install_script_download_link'] = \
             install_script_download_link
         update_agent_runtime_properties(cloudify_agent)
+        utils.update_agent_record(agent_name, AgentState.CREATED)
 
 
 @operation
 @create_agent_config_and_installer
 def configure(cloudify_agent, installer, **_):
-    ctx.logger.info('Configuring Agent {0}'.format(cloudify_agent['name']))
-    installer.configure_agent()
+    agent_name = cloudify_agent['name']
+    ctx.logger.info('Configuring Agent {0}'.format(agent_name))
+    utils.update_agent_record(agent_name, AgentState.CONFIGURING)
+    try:
+        installer.configure_agent()
+    except CommandExecutionError, e:
```

Comment:
don't use the old/deprec syntax (stops working in py3), use `except CommandExecutionError as e` instead